packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -105,7 +105,7 @@ module.exports = (
         return await ASTPromiseMap.get(cacheKey)
       } else {
         const ASTGenerationPromise = new Promise(async resolve => {
-          if (process.env.NODE_ENV === `production` && !fileNodes) {
+          if (!fileNodes) {
```

Comment:
My bad on this — the intent was that during production builds, we would generate the fileNodes array only once (as it doesn't change) but that during development, we'd keep regenerating it as file nodes get added/removed potentially all the time.

So the needed change here would be to add an `else` that generates the fileNodes every time during development.